### PR TITLE
[improve][doc] Document configuration added by PIP-145

### PR DESCRIPTION
**Pull request by andrasbeni**
Master Issue: #14505

### Motivation

Documentation for the new configuration properties was missing when the feature was merged. 

### Modifications

Added the new configuration properties to standalone.conf and broker.conf. Also, added documentation to site2/docs/reference-configuration.md .

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation  
- [X] `doc`

### CODE REVIEW

File: site2/docs/reference-configuration-broker.md
Comment (Anonymitaet) [inline site2/docs/reference-configuration-broker.md:2332]: Hi @SignorMercurio 

before https://github.com/apache/pulsar/pull/16865 gets merged and takes effect, do we allow users to update configuration md files manually? I think no, right? 

Because after your PR gets merged and takes effect, the new changes will be synced and shown on the website, right？

**Review by Anonymitaet — CHANGES_REQUESTED** [review]

File: site2/docs/reference-configuration-broker.md
Comment (SignorMercurio) [reply site2/docs/reference-configuration-broker.md:2332]: Totally right.

Comment (Anonymitaet) [reply site2/docs/reference-configuration-broker.md:2332]: @andrasbeni thanks for your contribution! We're trying to generate configuration doc from code automatically to avoid manual work https://github.com/apache/pulsar/issues/13453, could you please revert the changes to `.md` files?

Comment (andrasbeni) [reply site2/docs/reference-configuration-broker.md]: Thanks for the review. Changes to the .md files have been reverted.

**Review by momo-jun — APPROVED** [review]
@andrasbeni thanks for adding docs for #14804.

File: site2/docs/reference-configuration-broker.md
Comment (andrasbeni) [reply site2/docs/reference-configuration-broker.md]: @Anonymitaet, as I can see, #16865 is not going to be merged. What can I do now to get this PR merged?

File: conf/broker.conf
Comment (Anonymitaet) [inline conf/broker.conf:607]:
```suggestion
# Setting it to true means the broker only sends the list of matching topics to the clients.
```

**Review by Anonymitaet — COMMENTED** [review]
@freeznet 
could you please review this PR from the technical perspective? Thank you!

**Review by freeznet — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

File: site2/docs/reference-configuration-broker.md
Comment (Anonymitaet) [reply site2/docs/reference-configuration-broker.md]: @andrasbeni we can merge this PR after all tests pass

**Pull request merged**
